platform/linux-generic/odp_ipsec.c

```diff
@@ -566,17 +601,44 @@ static int ipsec_in_esp(odp_packet_t *pkt,
 	param->auth_iv_ptr = state->iv;
 
 	state->esp.aad.spi = esp.spi;
-	state->esp.aad.seq_no = esp.seq_no;
 	state->in.seq_no = odp_be_to_cpu_32(esp.seq_no);
 
+	/* AAD with ESN */
+	if (ipsec_sa->esn) {
+		state->in.seq64 = ipsec_compute_esn(ipsec_sa, state->in.seq_no);
+		if (ipsec_sa->combined_mode)
```

Comment:
The check for combined mode is not really necessary here. It was not done for the non-ESN case before and still isn't. There is not much harm in doing a possibly unnecessary write to the state structure that is written here anyway, but the branch for the combined_mode test is potentially unpredictable. And the same comment for the outbound case.

Reply:
I will check the feasibility of not using "combined_mode" flag here and in outbound case as well.

Reply:
Yes, everything seems fine with out the "combined_mode" flag check.
After removing these checks, it seems that there is no significance of having "combined_mode" flag, only the "insert_seq_hi" flag seems sufficient, so removed this "combined_mode" as well.